### DISCUSSION TITLE
New Apptainer Version doesen´t work?????

### DISCUSSION THREAD

**Discussion by DirkDali** [nextfoam/baram#91] [General]
Hello
Apptainer version 24.0.1 works fine.
Apptainer version 24.2.0 doesn't work at all.
Do you have any idea what's wrong?

![image](https://github.com/nextfoam/baram/assets/163728849/d6c36b84-ca26-42b7-a673-622037aab90d)

**Comment by kdg5424** [comment]
Hello @DirkDali. This is Donggyu.
Could you let me know your OS?
We'll try your own OS.
Thanks

Donggyu kim

**Reply by DirkDali** [reply]
Linux Mint Cinnamon 21.3
Your Apptainer installation on the website is no longer available?

**Reply by kdg5424** [reply]
Hello Dirk.
For the Linux Mint, we recommend you to install with Python virtual environment.
We don't have the enough human resources for supporting apptainer. So, we took apptainer off the site.
You can use BARAM as Python virtual environment in Linux Mint without any problems.
Do you have any problems for using Python virtual environment?

**Reply by DirkDali** [reply]
OK, it was a bit difficult because Linux Mint runs with Python 3.10 but I got it to work.

Thank you very much!

**Reply by DirkDali** [reply]
![image](https://github.com/nextfoam/baram/assets/163728849/b51f2264-6f2f-4fe4-8d0a-faf8fe55be2f)

Oh guys, you guys aren't making it easy. Baram Mesh is running fine. Baram Flow isn't running at all, see terminal!
Do you have any idea what's wrong?

**Reply by jiban** [reply]
Hi, Dirk.
Thank you for the capture.
It has to do with your *locale*, where ","(comma) is used for floating point.
You may run BARAM with a declaration of `LC_NUMERIC` like following.
(And you can insert the declaration into `baramMesh.sh` and `baramFlow.sh`, which is also shown in following capture.
```bash
jake@jake-Virtual-Machine:~/workspace/baram$ ls
baramFlow     build.gradle  gradlew.bat  misc         requirements.txt  solvers
baramFlow.sh  convertUi.py  INSTALL.md   __pycache__  resource.qrc      solvers_linux_24.2.1_20240620.tar.xz
baramMesh     gradle        libbaram     PyFoam       resource_rc.py    venv
baramMesh.sh  gradlew       LICENSE      README.md    resources         widgets
jake@jake-Virtual-Machine:~/workspace/baram$ ./baramFlow.sh 
[2024-07-04 12:40:12,654][root] ==> Uncaught exception
Traceback (most recent call last):
  File "/home/jake/workspace/baram/baramFlow/view/main_window/start_window.py", line 49, in _createProject
    app.plug.createProject(self._projectSelector)
  File "/home/jake/workspace/baram/baramFlow/app_plug_in.py", line 35, in createProject
    self._caseWizard = CaseWizard(parent, path)
  File "/home/jake/workspace/baram/baramFlow/view/case_wizard/case_wizard.py", line 34, in __init__
    self._db = coredb.createDB()
  File "/home/jake/workspace/baram/baramFlow/coredb/coredb.py", line 61, in createDB
    __instance = _CoreDB()
  File "/home/jake/workspace/baram/baramFlow/coredb/coredb.py", line 125, in __init__
    self._xmlSchema = etree.XMLSchema(etree=xsdTree)
  File "src/lxml/xmlschema.pxi", line 89, in lxml.etree.XMLSchema.__init__
lxml.etree.XMLSchemaParseError: Element '{http://www.w3.org/2001/XMLSchema}element', attribute 'default': [facet 'minExclusive'] The value '0.5' must be greater than '0'., line 1528
jake@jake-Virtual-Machine:~/workspace/baram$ LC_NUMERIC="C" ./baramFlow.sh 
jake@jake-Virtual-Machine:~/workspace/baram$ 
jake@jake-Virtual-Machine:~/workspace/baram$ 
jake@jake-Virtual-Machine:~/workspace/baram$ vi ./baramFlow.sh 
jake@jake-Virtual-Machine:~/workspace/baram$ tail ./baramFlow.sh 
    DIR=$( cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1 && pwd )
    SOURCE=$(readlink "$SOURCE")
    [[ $SOURCE != /* ]] && SOURCE=$DIR/$SOURCE # if $SOURCE was a relative symlink, we need to resolve it relative to the path where the symlink file was located
done

cd -P "$( dirname "$SOURCE" )" >/dev/null 2>&1

source venv/bin/activate

LC_NUMERIC="C" python -m baramFlow.main
jake@jake-Virtual-Machine:~/workspace/baram$ 
```
![Screenshot 2024-07-04 133257](https://github.com/nextfoam/baram/assets/1674177/7a6e8263-513f-4d9f-8cb5-858e70a9fe6f)

**Reply by jiban** [reply]
By the way, I will update BARAM itself to avoid  this problem.
You will not need this declaration from the next release.

**Reply by DirkDali** [reply]
OK, now I can start BaramFlow and open files. Your tip was very good! OLE!!
Unfortunately, the program hangs after I press Start Calculation, and it hangs here:
![image](https://github.com/nextfoam/baram/assets/163728849/71306d4e-64a2-4b44-9841-0c8ee5c3c3b6)

There is no file "baramd" in the directory: solvers/openfoam/bin
![image](https://github.com/nextfoam/baram/assets/163728849/a2128c63-6135-431a-be94-cf7dbfc765f2)

**Reply by kdg5424** [reply]
Hello Dirk.
Could you show me your terminal status?
I want to know what error is happened.

Thanks

**Reply by jiban** [reply]
@DirkDali You may need to compile `baramd` on your host like following.
```bash
gcc -o solvers/openfoam/bin/baramd misc/baramd.c
```
Please refer to this [baram page](https://baramcfd.org/installation/2024/04/15/installationPythonEnv-post/#compile-daemonizer--only-for-linux-and-macos-)

**Reply by DirkDali** [reply]
Great, it works! I forgot to compile.
Thanks for your help!
Dirk

**Reply by jiban** [reply]
I'm happy to hear that it works.
Enjoy your BARAM!

-- Jake

**Comment by bosung-gotocloud** [comment]
I think, the GLX related issues with Apptainer (or container) are related to various reasons. So, I asked to ChatGPT about issue.
https://chatgpt.com/share/44b83276-c9c1-47ff-9599-94a0ce0b15f3
MESA libraries and some GL related libraries are installed in the apptainer, however there maybe some mismatches in your environment.

Check the glx is working in your host OS and in the apptainer together
$ glxinfo 
In the container, 
$ vglrun glxinfo

There should be 'DIRECT RENDERING YES' in both cases

Issues continue in the Apptainer, we will check, If the previouse versions are working well, I recommend to copy `/opt/baram` directorie from the new apptainer to the old apptainer and try to run again.
